src/state/slices/userSettings.js

```diff
@@ -0,0 +1,54 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { PARAMS, setQueryParm, useQueryParam } from '../../utils/queryParam';
+
+function trycatch(func, fail) {
+  try {
+    return func();
+  } catch (e) {
+    return fail;
+  }
+}
+
+const SETTINGS_STORAGE_KEY = 'globalSettings';
+
+const defaultState = { expertMode: true, gameMode: 'fractals', language: 'en' };
+// eslint-disable-next-line react-hooks/rules-of-hooks
+const gameModeParam = useQueryParam({ key: PARAMS.GAMEMODE });
+
+// settings grabbed from the local storage. Not the final settings, there are overrides!
+const localStorageSettings = JSON.parse(localStorage.getItem(SETTINGS_STORAGE_KEY));
+export const loadedSettings = {
+  ...defaultState,
+  ...trycatch(() => localStorageSettings, {}), // override default state with potentially saved localStorage variables
```

Comment:
The JSON.parse needs to be in this callback for errors to be caught!

Reply:
oh my god - it used to be there, but I tried some things and forgot to put it back